### PR TITLE
add ottolenghi books scraper

### PR DESCRIPTION
**Pull request by tomtjes**
Adds support for books.ottolenghi.co.uk
Some books have slightly different formatting. I have tested this with:
- Sweet
- Simple
- Shelf Love
- Plenty More
- Plenty
- Flavor
- Extra Good Things

I currently do not have access to:
- Nopi
- Jerusalem
- Falastin
- Cookbook

### CODE REVIEW

**Comment by strangetom** [comment]
Thanks @tomtjes 

Confirmed working with Falastin too.

It should be noted that access to the recipes on this site requires logging in (but I don't think that affects anything in this PR).

File: README.rst [hunk 1]
```diff
@@ -121,6 +121,7 @@ Scrapers available for:
 - `https://www.bodybuilding.com/ <https://www.bodybuilding.com/>`_
 - `https://bonappetit.com/ <https://bonappetit.com>`_
 - `https://bongeats.com/ <https://bongeats.com/>`_
+- `https://books.ottolenghi.co.uk <https://books.ottolenghi.co.uk/>`_
```
Comment (jayaddison) [inline README.rst:124]:
```suggestion
- `https://books.ottolenghi.co.uk <https://books.ottolenghi.co.uk/>`_ (*)
```

Nitpick: let's add an asterisk annotation to this scraper to note that it runs on non-public HTML content (a few other scrapers have this annotation for similar reasons).

**Review by jayaddison — APPROVED** [review]
Looks good to me too :+1: 

The number-parsing code is neat; perhaps that will be re-usable either within `get_yields` or more generally at some point.

Thank you, @tomtjes!

**Comment by jayaddison** [comment]
Don't worry about the merge conflict that appears here.. yet; I'm performing some release tasks on the `main` branch, and I _think_ that that may be resolved automatically soon.

**Comment by jayaddison** [comment]
This is now available in versions 14.59.0 and 15.1.0, as published to PyPi and GitHub.  Thank you again!

**Pull request merged**